src/ReplicatedStorage/Client/InstanceComponents/Misc/Sprint.lua

```diff
@@ -0,0 +1,63 @@
+local Players = game:GetService("Players")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local ContextActionService = game:GetService("ContextActionService")
+local Trove = require(ReplicatedStorage.Packages.Trove)
+local Comm = require(ReplicatedStorage.Packages.Comm)
+
+local Sprint = {}
+Sprint.__index = Sprint
+
+function Sprint.new(character: Model)
+	local clientComm = Comm.ClientComm.new(character, false, "Sprint")
+	local self = setmetatable({}, Sprint)
+
+	self._comm = clientComm
+	self._trove = Trove.new()
+	self._trove:Add(self._comm)
```

Comment:
Couldn't we use Trove:Construct?

Reply:
Probably. I didn't know about trove construct until a bit ago, and planet's code is partially based on mine which is where this comes from.